pkg/encoding/scheme.go

```diff
@@ -16,134 +16,113 @@ limitations under the License.
 
 package encoding
 
+// Don't edit this file directly. It is generated by scheme.sh.
 import (
-	"os"
-
-	admissionv1beta1 "k8s.io/api/admission/v1beta1"
-
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
 	admissionregistrationv1alpha1 "k8s.io/api/admissionregistration/v1alpha1"
 	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
-
+	apiserverinternalv1alpha1 "k8s.io/api/apiserverinternal/v1alpha1"
 	appsv1 "k8s.io/api/apps/v1"
 	appsv1beta1 "k8s.io/api/apps/v1beta1"
 	appsv1beta2 "k8s.io/api/apps/v1beta2"
-
 	authenticationv1 "k8s.io/api/authentication/v1"
+	authenticationv1alpha1 "k8s.io/api/authentication/v1alpha1"
 	authenticationv1beta1 "k8s.io/api/authentication/v1beta1"
-
 	authorizationv1 "k8s.io/api/authorization/v1"
 	authorizationv1beta1 "k8s.io/api/authorization/v1beta1"
-
 	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
 	autoscalingv2beta1 "k8s.io/api/autoscaling/v2beta1"
-
+	autoscalingv2beta2 "k8s.io/api/autoscaling/v2beta2"
 	batchv1 "k8s.io/api/batch/v1"
 	batchv1beta1 "k8s.io/api/batch/v1beta1"
-	batchv2alpha1 "k8s.io/api/batch/v2alpha1"
-
+	certificatesv1 "k8s.io/api/certificates/v1"
+	certificatesv1alpha1 "k8s.io/api/certificates/v1alpha1"
 	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
-
+	coordinationv1 "k8s.io/api/coordination/v1"
+	coordinationv1beta1 "k8s.io/api/coordination/v1beta1"
 	corev1 "k8s.io/api/core/v1"
-
+	discoveryv1 "k8s.io/api/discovery/v1"
+	discoveryv1beta1 "k8s.io/api/discovery/v1beta1"
+	eventsv1 "k8s.io/api/events/v1"
 	eventsv1beta1 "k8s.io/api/events/v1beta1"
-
 	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
-
-	imagepolicyv1alpha1 "k8s.io/api/imagepolicy/v1alpha1"
-
+	flowcontrolv1 "k8s.io/api/flowcontrol/v1"
+	flowcontrolv1beta1 "k8s.io/api/flowcontrol/v1beta1"
+	flowcontrolv1beta2 "k8s.io/api/flowcontrol/v1beta2"
+	flowcontrolv1beta3 "k8s.io/api/flowcontrol/v1beta3"
 	networkingv1 "k8s.io/api/networking/v1"
-
+	networkingv1alpha1 "k8s.io/api/networking/v1alpha1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	nodev1 "k8s.io/api/node/v1"
+	nodev1alpha1 "k8s.io/api/node/v1alpha1"
+	nodev1beta1 "k8s.io/api/node/v1beta1"
+	policyv1 "k8s.io/api/policy/v1"
 	policyv1beta1 "k8s.io/api/policy/v1beta1"
-
 	rbacv1 "k8s.io/api/rbac/v1"
 	rbacv1alpha1 "k8s.io/api/rbac/v1alpha1"
 	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
-
+	resourcev1alpha2 "k8s.io/api/resource/v1alpha2"
+	schedulingv1 "k8s.io/api/scheduling/v1"
 	schedulingv1alpha1 "k8s.io/api/scheduling/v1alpha1"
-
-	settingsv1alpha1 "k8s.io/api/settings/v1alpha1"
```

Comment:
Should we somehow support api's that have been removed?

#10

Reply:
I think it is fine to not support apis that have been removed. But we should
* Build and publish releases for all recent kubernetes versions (1.6+)
* Can you add instructions about how to build the tool for an specific version of k8s in README using the script? It can be a follow up PR.

Reply:
> Build and publish releases for all recent kubernetes versions (1.6+)

I think it would be possible to use a script to copy the deleted apiversion to the current repository, similar to what this PR currently does.

> Can you add instructions about how to build the tool for an specific version of k8s in README using the script? It can be a follow up PR.

Of course, I've moved the initialisation of scheme to main, which allows different versions to co-exist, and I'll try to improve the documentation later.

Reply:
while it would be possible to use a script to copy the deleted apiversion to the current repository, we still need to make decisions about how far back we want to support. We can start a discussion in the other issue.